### PR TITLE
Add CI

### PR DESCRIPTION
**Pull request by W95Psp**
Related to #4

### CODE REVIEW

File: .github/workflows/specs.yml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+name: Test extractions for `hacpsec/specs`
+
+on:
+  push:
+    branches:
+    - '*'
```
Comment (franziskuskiefer) [inline .github/workflows/specs.yml:6]: Let's run this on PRs and on demand as well.
At some point this shouldn't run on branches anymore, but that's ok for now.

```suggestion
  pull_request:
  workflow_dispatch:
```

File: .github/workflows/specs.yml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    - uses: cachix/cachix-action@v12
+      with:
+        name: hacspec
+        authToken: '${{ secrets.CACHIX_AUTH_TOKEN }}'
+    - name: Build
+      run: nix build
+
+    - name: Install
+      run: |
+        nix profile install nixpkgs#yq
```
Comment (franziskuskiefer) [inline .github/workflows/specs.yml:23]: Please document in #4 that this is using nix for now and that we need to add this without nix as well.

File: .github/workflows/compile_engine.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+jobs:
+  release:
+    if: startsWith(github.ref, 'refs/tags/')
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - macos-latest
+          - ubuntu-latest
+          # - windows-latest
```
Comment (franziskuskiefer) [inline .github/workflows/compile_engine.yml:17]: Windows isn't working?

File: .github/workflows/compile_engine.yml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      contents: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - uses: ocaml/setup-ocaml@v2
+        with:
+          ocaml-compiler: ${{ matrix.ocaml-compiler }}
+
+      - uses: actions-rs/toolchain@v1
```
Comment (franziskuskiefer) [inline .github/workflows/compile_engine.yml:31]: `actions-rs` is unmaintained. You can use `dtolnay/rust-toolchain` if you want to use a toolchain action.

File: .github/workflows/compile_engine.yml [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+        id: rust_toolchain
+        with:
+          toolchain: stable
+          
+      - run: cargo install --path cli
```
Comment (franziskuskiefer) [inline .github/workflows/compile_engine.yml:36]: Why do we need the cli when building the engine?

File: .github/workflows/compile_engine.yml [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+      - run: opam install . --deps-only
```
Comment (franziskuskiefer) [inline .github/workflows/compile_engine.yml:38]: How do we compile for m1 here? Is there some `--target` in opam/dune?

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Thanks!
Generally lgtm with a few nits. The unmaintained dependency is the main point that needs to get fixed.
Can you also update #4 with what's in here and what still needs to be done?

File: .github/workflows/specs.yml (hunk 2 above)
Comment (W95Psp) [reply .github/workflows/specs.yml:23]: (I edited #4)

File: .github/workflows/compile_engine.yml (hunk 3 above)
Comment (W95Psp) [reply .github/workflows/compile_engine.yml:17]: It was not due to incompatibility of some library, but I might I have fixed that, will check

File: .github/workflows/compile_engine.yml (hunk 5 above)
Comment (W95Psp) [reply .github/workflows/compile_engine.yml:56]: The Rust part (aka the cli) gives the types to the ML part, it provides a binary `cargo-circus-export-json-schema` (or something) that outputs JSON schema, which is then processed by a script (the dirty JS one) called by dune to create a ML module of input types

File: .github/workflows/compile_engine.yml (hunk 6 above)
Comment (W95Psp) [reply .github/workflows/compile_engine.yml:58]: I have absolutely no idea :/
Also, the binaries I produce are not static, and it seems not that easy to make OCaml output fully static binaries...

That's why for now I just compile to JS, I think that's a good fallback solution (`nodejs` is available basically on every plateform)

File: .github/workflows/compile_engine.yml (hunk 3 above)
Comment (W95Psp) [reply .github/workflows/compile_engine.yml]: So, turns out Opam is not able to install `gmp` or `pkgconf`, I'll mark that as a TODO in the issue #4

**Comment by karthikbhargavan** [comment]
Let's merge this soon and then improve over time.

File: .github/workflows/compile_engine.yml (hunk 6 above)
Comment (franziskuskiefer) [reply .github/workflows/compile_engine.yml:58]: ok, let's use this for now and discuss how to do this.

File: .github/workflows/specs.yml (hunk 2 above)
Comment (franziskuskiefer) [reply .github/workflows/specs.yml:22]: We'll need to make sure that things work for everyone. But let's discuss and do something in a follow up.

**Review by franziskuskiefer — APPROVED** [review]
Let's get this.

**Pull request merged**